iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant.inl

```diff
@@ -231,33 +236,36 @@ inline const T* variant<Types...>::get() const noexcept
     {
         return nullptr;
     }
-    else
-    {
-        return static_cast<const T*>(static_cast<const void*>(m_storage));
-    }
+    // AXIVION Next Construct AutosarC++19_03-A5.2.3 : avoid code duplication
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-const-cast)
+    return static_cast<const T*>(static_cast<const void*>(m_storage));
```

Comment:
Well, I disagree here. IMHO it's a cleaner solution to use a two stage `static_cast` first to `void*` and then to the type. The reason for this is that `reinterpret_cast` should only be used if really necessary. Additionally, I think it would allow us to make `variant` easier `constexpr`-able.

Reply:
@elBoberido Yeah, that's a good idea. I wouldn't call it `iox_reinterpret_cast`, though :sweat_smile: Maybe `cast__via_void` or something like that.

Reply:
Some food for thought: https://stackoverflow.com/questions/1863069/casting-via-void-instead-of-using-reinterpret-cast

Reply:
@mossmaurice how about `iox_type_cast`?

Reply:
@elBoberido I like the name! I'll add it to `helplets.hpp` on this branch.

Reply:
Created an issue #1498